storage/ksql/src/main/java/io/apicurio/registry/storage/impl/ksql/sql/KafkaSQLSink.java

```diff
@@ -233,4 +264,16 @@ private boolean isGlobalRules(String artifactId) {
         return artifactId.equals(KafkaSqlRegistryStorage.GLOBAL_RULES_ID);
     }
 
+    public long toGlobalId(long offset, int partition) {
+        return getBaseOffset() + (offset << 16) + partition;
+    }
+
+    // just to make sure we can always move the whole system
+    // and not get duplicates; e.g. after move baseOffset = max(globalId) + 1
+    public long getBaseOffset() {
+        //TODO
+        //        return Long.parseLong(properties.getProperty("storage.base.offset", "0"));
+        return 0;
+    }
+
```

Comment:
We may need to revisit this for multi-tenancy if we go that way.  Fine for now though.  I wonder if we could use 10 or 12 bits when shifting to give us more headroom for the offset.  The less we bitshift the offset the fewer partitions we can have, but right now we can have 65k partitions which seems high.  Bitshifting by 10 would give us 1024 partitions and more headroom before we overran the integer.  Overrunning the integer is a concern for the ccompat API, which uses an integer for the globalId instead of a long.

Thoughts @Apicurio/developers ?